src/main/java/pl/debuguj/parkingspacessystem/services/ParkingSpaceManagementServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package pl.debuguj.parkingspacessystem.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.debuguj.parkingspacessystem.dao.ParkingSpaceDao;
+import pl.debuguj.parkingspacessystem.exceptions.CarRegisteredInSystemException;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+import pl.debuguj.parkingspacessystem.domain.ParkingSpace;
+import pl.debuguj.parkingspacessystem.exceptions.ParkingSpaceNotFoundException;
+
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.Optional;
+
+/**
+ * Created by grzesiek on 09.10.17.
+ */
+@Service
+public class ParkingSpaceManagementServiceImpl implements ParkingSpaceManagementService {
```

Comment:
to nie musi być publiczna klasa. Poza tym, po co ci osobny interfejs dla pojedynczej implementacji?

Reply:
Interfejs jest mi potrzebny w przypadku, gdybym potrzebował nowej implementacji. To dość elastyczne rozwiązanie. Przy serwisach mam nawyk tak robić. Przy pojedynczej implementacji może to być zbędne, ale na dłuższą metę przeważnie było pomocne.